assets/js/newsletter.js

```diff
@@ -0,0 +1,15 @@
+---
+# Empty front matter to enable Jekyll processing
+---
+$("#newsletter-modal").on('shown.bs.modal', function(w, d, s, o, f, js, fjs) {
```

Comment:
If you only want to run the code once the modal is opened, the function should be wrapped in another function. As it is now, the function is called immediately, returning `undefined` and thus no callback is invoked for the `shown.bs.modal` event.

Reply:
Can you suggest a specific change? I'm a bit lost in JS. Is it just wrapping in another `function() {`...`}`?

Reply:
I assume you are trying to create a closure with specific variables without relying on global object.
```js
$("#newsletter-modal").on('shown.bs.modal', function(w, d, s, o, f, js, fjs) {
  return function onShowCallback() {
    console.log('this code is run on dialog showing')
  }
}());  // these closed brackets cause function to be called immediately and return the callback that becomes argument for 'on' call
```

I personally have no issue with using global object variables like `window, document, etc.`. But maybe you have a specific reason for this.

Reply:
As I said earlier, JS is a Spanish Village to me :D. I'd be happy with either solution you propose (both being more experienced in JS). Should we do the closure-thingy or other approach?